extensions/amp-analytics/0.1/requests.js

```diff
@@ -318,6 +345,73 @@ export class RequestHandler {
     }
     return s.join('&');
   }
+
+  /**
+   * Handle batchInterval
+   */
+  initBatchInterval_() {
+    if (!this.batchInterval_) {
+      return;
+    }
+
+    if (this.maxDelay_) {
+      // Ignore maxDelay in presence of batchInterval
+      // TODO: we can remove the restriction upon requests
+      user().error(TAG,
+          'maxDelay value will not be respected with batchInterval defined');
+      this.maxDelay_ = 0;
+    }
+
+    if (isArray(this.batchInterval_)) {
+      // Support batchInterval in array with absolute time to send request
+      const batchInterval = [];
+      for (let i = 0; i < this.batchInterval_.length; i++) {
+        const current = this.batchInterval_[i];
+        const prev = (i > 0) ? this.batchInterval_[i - 1] : 0;
+        user().assert(isFiniteNumber(current) && Number(current) >= 0,
+            `Invalid batchInterval value: ${this.batchInterval_}, ` +
+            'interval must be number greater than 0.');
+        const interval = (current - prev) * 1000;
```

Comment:
So batchInterval doesn't contain intervals it contains absolute times for firing? eg. [10, 20, 30, 40] --> fire at 10 seconds from load, then 10 seconds after that, 10 seconds after that, etc. This would be equivalent to maxDelay: 10 ?

I think it makes more sense to config with intervals instead of absolute times. So for example a config which increases the interval by 10s every tick:  maxDelay: [10, 20, 30, 40]  Will fire at absolute times from load: 10s, 30s, 50s, 90s, etc. This also means it lines up with what maxDelay specifies.